static/src/js/util/retrieveVariablesRequest.js

```diff
@@ -0,0 +1,89 @@
+/**
+ *
+  * @param {Object} data collection metadata
+  * @returns {Object} data with the additional variables to be extracted
+ */
+const formatCollectionObject = (data) => {
+  // If the input object contains multiple collections
+  if (data.collections) {
+    const formattedData = {}
+
+    const { collections } = data
+    const { items } = collections
+    const { variables } = items[0]
+    formattedData.collection = {
+      variables
+    }
+
+    return formattedData
+  }
+
+  return data
+}
+
+/**
+ * Retrieves the variables of a collection if there are more than `maxCmrPageSize`
+  * @param {Object} variablesObj: contains cusor and item from the original graphQlReuqest
+  * @param {Object} requestParams: the variable request params
+  * @param {Object} graphQlRequestObject: used to make additional graphql requests
+  * @returns {Object} retrieves the rest of the variables for a given collection
+ */
+export const retrieveVariablesRequest = async (
+  variablesObj,
+  requestParams,
+  graphQlRequestObject
+) => {
+  // Pull out cursor and already retrieved items from the variablesObj
+  const { cursor, items } = variablesObj
+
+  let collectedItems = items
+  let nextCursor = cursor
+  while (nextCursor !== null) {
+    const varsGraphQuery = `
+        query GetCollection(
```

Comment:
Are there alternatives to using the GetCollection query? Can you just directly query for variables instead?

Reply:
Great catch, you absolutely should query variables directly. As is this is doubling the amount of queries that hit CMR from what you actually need

Reply:
Can we do that? We need a reference for which variables to get `cmr-graphql` doesn't allow us to query variables by the collection concept-id. CMR itself doesn't support that in the documentation either `https://cmr.earthdata.nasa.gov/search/site/docs/search/api.html#variable`

Reply:
I just tried with `keyword` that does work we should use that to search for the variables directly